lib/mappers/pa_mapper.py

```diff
@@ -0,0 +1,122 @@
+from dplaingestion.selector import exists, getprop
+from dplaingestion.mappers.dublin_core_mapper import DublinCoreMapper
+from dplaingestion.utilities import iterify
+from dplaingestion.selector import exists, setprop
+from urlparse import urlparse
+from akara import logger
+
+
+class PAMapper(DublinCoreMapper):
+
+    DCMI_TYPES = {
+        "Collection",
+        "Dataset",
+        "Event",
+        "Image",
+        "InteractiveResource",
+        "MovingImage",
+        "PhysicalObject",
+        "Service",
+        "Software",
+        "Sound",
+        "StillImage",
+        "Text"
+    }
+
+    def __init__(self, provider_data):
+        super(PAMapper, self).__init__(provider_data)
+
+    def map_relation(self):
+        prop = "relation"
+        if exists(self.provider_data, prop):
+            relations = iterify(self.provider_data.get(prop))
+            self.update_source_resource({"collection": {"title": relations[0]}})
+            if len(relations) > 1:
+                self.update_source_resource({"relation": relations[1:]})
+
+    def map_rights(self):
+        prop = "rights"
+        if exists(self.provider_data, prop):
+            rights_uri = ""
+            rights = self.provider_data.get(prop)
+            try:
+                if rights.startswith("http"):
+                    rights_uri = urlparse(rights).geturl()
+            except Exception as e:
+                logger.warn("Unable to parse rights URI: %s\n%s" % (rights, e))
+
+            if rights_uri:
+                self.mapped_data.update({"rights": rights_uri})
+            else:
+                self.update_source_resource({"rights": rights})
+
+    def map_type(self):
+        prop = "type"
+        if exists(self.provider_data, prop):
+            types = iterify(self.provider_data.get(prop))
+            non_dcmi_types = [type for type in types if type not in self.DCMI_TYPES]
+            if len(non_dcmi_types) > 0:
+                self.update_source_resource({"format": non_dcmi_types})
+            dcmi_types = [type for type in types if type in self.DCMI_TYPES]
+            if (len(dcmi_types)) > 0:
+                self.update_source_resource({"type": dcmi_types})
+
+    def map_spatial(self):
+        prop = "coverage"
+        if exists(self.provider_data, prop):
```

Comment:
The property can exist, but could theoretically be `None` or empty (evaluating to `False`), in which case we wouldn't want it to assign it. The code below will assign it. Would it be safer to get the property (defaulting to None) and then say `if coverage: ...`?

Same goes for other functions like `map_contributor()`, `map_identifier()`, etc.